src/nomad_catalysis/schema_packages/schema.py

```diff
@@ -0,0 +1,424 @@
+        type=str,
+        shape=[],
+        description="""
+          A description of the method used to measure the surface area of the sample.
+          """,
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(
+                suggestions=[
+                    'BET',
+                    'H2-TPD',
+                    'N2O-RFC',
+                    'Fourier Transform Infrared Spectroscopy (FTIR)' ' of adsorbates',
+                    'unknown',
+                ]
+            ),
+        ),
+    )
+
+    dispersion = Quantity(
+        type=np.float64,
+        shape=[],
+        description="""
+        The dispersion of the catalyst in %.
+        """,
+        a_eln=dict(component='NumberEditQuantity'),
+    )
+
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+
+
+class CatalystSample(CompositeSystem, Schema):
+    m_def = Section(
+        label='Catalyst Sample',
+        categories=[CatalysisElnCategory],
+    )
+
+    preparation_details = SubSection(
+        section_def=Preparation, a_eln=ELNAnnotation(label='Preparation Details')
+    )
+
+    surface = SubSection(
+        section_def=SurfaceArea, a_eln=ELNAnnotation(label='Surface Area')
+    )
+
+    storing_institution = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        The institution at which the sample is stored.
+        """,
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(
+                suggestions=[
+                    'Fritz-Haber-Institut Berlin / Abteilung AC',
+                    'Fritz-Haber-Institut Berlin / ISC',
+                    'TU Berlin / BasCat',
+                ]
+            ),
+        ),
+    )
+
+    catalyst_type = Quantity(
+        type=str,
+        shape=['*'],
+        description="""
+          A classification of the catalyst type.
+          """,
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(
+                suggestions=[
+                    'bulk catalyst',
+                    'supported catalyst',
+                    'single crystal',
+                    'metal',
+                    'oxide',
+                    '2D catalyst',
+                    'other',
+                    'unkown',
+                ]
+            ),
+        ),
+        links=['https://w3id.org/nfdi4cat/voc4cat_0007014'],
+    )
+
+    form = Quantity(
+        type=str,
+        shape=[],
+        description="""
+          classification of physical form of catalyst
+          """,
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(suggestions=['sieve fraction', 'powder', 'thin film']),
+        ),
+        links=['https://w3id.org/nfdi4cat/voc4cat_0000016'],
+    )
+
+    def normalize(self, archive, logger):
+        populate_catalyst_sample_info_to_results(archive, self, logger)
```

Comment:
Does this need to be a standalone function? I.e. are you using it somewhere else later? For now it really feels like a class method as you are only using it here and you are passing the self into it.

Reply:
I will use the function again in another class later in the next PR with the added keyword `reference=True`, but I will try to add a class function in between.